lib/wanda_web/auth/client/auth_client_http.ex

```diff
@@ -0,0 +1,59 @@
+defmodule WandaWeb.Auth.Client.HttpClient do
+  @moduledoc """
+  Http AuthClient implementation.
+  """
+
+  @behaviour WandaWeb.Auth.Client.AuthClient
+
+  alias WandaWeb.Auth.Client.AuthClient
+
+  require Logger
+
+  @impl AuthClient
+  def introspect_token(token) do
+    "#{auth_server_url()}/api/session/token/introspect"
```

Comment:
question:
doesn't this look like a typical `with :ok <- HTTPoison.post(), :ok <- get_response_body()`, etc kind of code?
I think I would find it more elixirsh.
And maybe having the `error_atom` and `error_log` directly in the failed clause as well. They are not reused in any other way.
For example, with the current code, if the `get_response_body` returns an error, it looks like it is printed twice.